internal/ai/ai.go

```diff
@@ -66,8 +69,29 @@ func (c *CachingAI) Chat() (string, error) {
 		}
 	}
 
-	response, err := c.ai.Chat()
-	if err != nil {
+	var response string
+	var err error
+	var lastErr error
+	retries := 0
+	maxRetries := c.config.MaxRetries
+
+	for retries <= maxRetries {
```

Comment:
I also have this util:

https://github.com/blacktop/ipsw/blob/10534629ea77e881124d6ed2dcbbdfbf4d5e9473/internal/utils/utils.go#L28